common/tools/dev-tool/src/commands/run/extract-api.ts

```diff
@@ -0,0 +1,150 @@
+      );
+      const newDtsRollupOptions: IConfigDtsRollup = {
+        ...extractorConfigObject.dtsRollup,
+        publicTrimmedFilePath: newPublicTrimmedPath,
+      };
+      const newDocModel: IConfigDocModel = {
+        ...extractorConfigObject.docModel,
+        enabled: true,
+        apiJsonFilePath: newApiJsonPath,
+      };
+      const newApiReport: IConfigApiReport = {
+        ...extractorConfigObject.apiReport,
+        enabled: true, // or should we disable? not sure if we need look at multiple api.md
+        reportFileName: newApiReportName,
+      };
+      const updatedConfigObject: IConfigFile = {
+        ...extractorConfigObject,
+        dtsRollup: newDtsRollupOptions,
+        docModel: newDocModel,
+        apiReport: newApiReport,
+        mainEntryPointFilePath: exports[exprt].types,
+      };
+      const extractorConfig = ExtractorConfig.prepare({
+        configObject: updatedConfigObject,
+        configObjectFullPath: apiExtractorJsonPath,
+        packageJsonFullPath: packageJsonPath,
+      });
+      const extractorResult: ExtractorResult = Extractor.invoke(extractorConfig, {
+        // Equivalent to the "--local" command-line parameter
+        localBuild: true,
+        // Equivalent to the "--verbose" command-line parameter
+        showVerboseMessages: true,
+      });
+      if (extractorResult.succeeded) {
+        log.debug(`API Extractor completed successfully for ${exprt} export`);
+      } else {
+        succeed = false;
+        log.error(
+          `API Extractor completed with ${extractorResult.errorCount} errors` +
+            ` and ${extractorResult.warningCount} warnings`
+        );
+      }
+    }
+  } else {
+    // normal extraction
+    const config = ExtractorConfig.prepare({
+      configObject: extractorConfigObject,
+      configObjectFullPath: apiExtractorJsonPath,
+      packageJsonFullPath: packageJsonPath,
+    });
+
+    const extractorResult: ExtractorResult = Extractor.invoke(config, {
+      // Equivalent to the "--local" command-line parameter
+      localBuild: true,
+      // Equivalent to the "--verbose" command-line parameter
+      showVerboseMessages: true,
+    });
+    if (extractorResult.succeeded) {
+      log.debug(`API Extractor completed successfully`);
+    } else {
+      succeed = false;
+      log.error(
+        `API Extractor completed with ${extractorResult.errorCount} errors` +
+        ` and ${extractorResult.warningCount} warnings`
+      );
+    }
+  }
+
+  // Add api.json files to zip archive
+  const unscopedPackageName = projectInfo.name.split("/")[1];
+  const reportTempDir  = path.join(projectInfo.path, "temp");
+  const files = (await readdir(reportTempDir)).filter(f => f.endsWith("api.json"));
+  const zip = new AdmZip();
+  for (const file of files) {
+    log.debug(`adding ${file} to zip archive`);
+    zip.addLocalFile(path.join(reportTempDir, file));
+  }
+  zip.writeZip(path.join(reportTempDir, `${unscopedPackageName}.zip`));
```

Comment:
will there be problem with `@azure` and `@azure-rest`? i.e., two packages with the same unscoped package name

Reply:
could we just add the scope, e.g. azure-openai or azure-rest-openai?

Reply:
Now I think about it again, this shouldn't matter as we only run the command within the package's folder there is only ever one single zip there.